File: getmsg.js
document.getElementById("contactForm").addEventListener("submit", async function(event) {
   event.preventDefault();

   const name = document.getElementById("name").value;
   const email = document.getElementById("email").value;
   const message = document.getElementById("message").value;

   const responseMessage = document.getElementById("responseMessage");

   try {
       const response = await fetch("http://127.0.0.1:5000/send-message", {
           method: "POST",
           headers: {
               "Content-Type": "application/json"
           },
           body: JSON.stringify({ name, email, message })
       });

       const result = await response.json();
       
       if (response.ok) {
           responseMessage.textContent = "✅ Message sent successfully!";
           responseMessage.style.color = "green";
           document.getElementById("contactForm").reset(); // Clear form
       } else {
           responseMessage.textContent = "❌ Error sending message!";
           responseMessage.style.color = "red";
       }
   } catch (error) {
       responseMessage.textContent = "❌ Server error, try again later.";
       responseMessage.style.color = "red";
   }
});

File: getmsg.py
from flask import Flask, request, jsonify
from flask_cors import CORS
import smtplib

app = Flask(__name__)
CORS(app)  # Allow frontend calls

# Replace with your email credentials
SENDER_EMAIL = "your_email@gmail.com"
SENDER_PASSWORD = "your_app_password"
RECEIVER_EMAIL = "your_email@gmail.com"

@app.route('/send-message', methods=['POST'])
def send_message():
    try:
        data = request.json
        name, email, message = data["name"], data["email"], data["message"]

        subject = f"New Message from {name}"
        body = f"From: {name} ({email})\n\n{message}"

        # Sending Email
        with smtplib.SMTP("smtp.gmail.com", 587) as server:
            server.starttls()
            server.login(SENDER_EMAIL, SENDER_PASSWORD)
            server.sendmail(SENDER_EMAIL, RECEIVER_EMAIL, f"Subject: {subject}\n\n{body}")

        return jsonify({"message": "Message sent successfully!"}), 200

    except Exception as e:
        return jsonify({"message": "Error sending message!", "error": str(e)}), 500

if __name__ == '__main__':
    app.run(debug=True)
